### PR TITLE
[MODULAR] Adult sound prefs

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
Adds in a new modified version of `playsounds` that functions the same but checks for the prefs of listeners, does not pierce walls, and does not play for ghosts. So far, this has been added for the adult interface items, but I do want to add this to interaction menu emotes in a future PR. 

There are two separate prefs: `ADULT` toy sounds, which disables the sounds for adult toys and `ADULT` sounds, which disables the sounds people make when they get very excited (and will control interaction menu sounds too.)

## How This Contributes To The Skyrat Roleplay Experience
It's important that we allow people to opt-out of hearing things that they wouldn't want to hear. This will hopefully cut down on administrative issues related to organic interface content.

## Proof of Testing

I can't post this on here, but I have a video I can share on discord.

<details>
<summary>Screenshots/Videos</summary>
  
image

image

</details>

## Changelog

:cl:
qol: Organic Interface sounds now have prefs associated with them.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/skyrat/erp_preferences.tsx [hunk 1]
```diff
@@ -14,6 +14,13 @@ export const erp_pref: FeatureToggle = {
   component: CheckboxInput,
 };
 
+export const erp_sounds_pref: FeatureToggle = {
+  name: 'ERP sounds',
+  category: 'ERP',
+  description: 'Toggles if you want to hear sounds from ERP mechanics.',
```
Comment (capsaicinz) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/skyrat/erp_preferences.tsx:20]: minor capitalization error? literally unplayable.

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_helpers/sounds.dm [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * vary - bool that determines if the sound changes pitch every time it plays.
+ * extrarange - modifier for sound range. This gets added on top of SOUND_RANGE.
+ * falloff_exponent - Rate of falloff for the audio. Higher means quicker drop to low volume. Should generally be over 1 to indicate a quick dive to 0 rather than a slow dive.
+ * frequency - playback speed of audio.
+ * channel - The channel the sound is played at.
+ * pressure_affected - Whether or not difference in pressure affects the sound (E.g. if you can hear in space).
+ * falloff_distance - Distance at which falloff begins. Sound is at peak volume (in regards to falloff) aslong as it is in this range.
+ * pref_to_check - the path of the pref that we want to check
+ */
+/proc/play_lewd_sound(atom/source, soundin, vol as num, vary, extrarange as num, falloff_exponent = SOUND_FALLOFF_EXPONENT, frequency = null, channel = 0, pressure_affected = TRUE, falloff_distance = SOUND_DEFAULT_FALLOFF_DISTANCE, use_reverb = TRUE, pref_to_check = /datum/preference/toggle/erp/sex_toy_sounds)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_helpers/sounds.dm]: When there's this many arguments, I feel like we're better off going multiline.
```suggestion
/proc/play_lewd_sound(
	atom/source,
	soundin,
	vol as num,
	vary,
	extrarange as num,
	falloff_exponent = SOUND_FALLOFF_EXPONENT,
	frequency = null,
	channel = 0,
	pressure_affected = TRUE,
	falloff_distance = SOUND_DEFAULT_FALLOFF_DISTANCE,
	use_reverb = TRUE,
	pref_to_check = /datum/preference/toggle/erp/sex_toy_sounds,
)
```

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/kinky_shocker.dm [hunk 3]
```diff
@@ -118,7 +118,7 @@
 		to_chat(user, span_danger("[target] doesn't want you to do that."))
 		return
 	deductcharge(cell_hit_cost)
-	playsound(loc, 'sound/weapons/taserhit.ogg', 70, 1, -1)
+	play_lewd_sound(loc, 'sound/weapons/taserhit.ogg', 70, 1, -1)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/kinky_shocker.dm:121]: I really wish I knew what all of these `1, -1` stood for.

Edit: I checked, all of the `1` (not `-1`) should actually be `TRUE`, it's a boolean. Not going to force you to do it in this PR though.

File: tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/skyrat/erp_preferences.tsx [hunk 4]
```diff
@@ -37,6 +44,13 @@ export const sextoy_pref: FeatureToggle = {
   component: CheckboxInput,
 };
 
+export const sextoy_sounds_pref: FeatureToggle = {
+  name: 'Sex toy sounds',
+  category: 'ERP',
+  description: 'toggles if you will hear sounds from sex toys.',
```
Comment (GoldenAlpharex) [inline tgui/packages/tgui/interfaces/PreferencesMenu/preferences/features/game_preferences/skyrat/erp_preferences.tsx]: I think Caps meant to comment here.
```suggestion
  description: 'Toggles if you will hear sounds from sex toys.',
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/kinky_shocker.dm (hunk 3 above)
Comment (softcerv) [reply modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/kinky_shocker.dm:121]: Yeah, I'll probably want to do that as a future PR

**Review by GoldenAlpharex — APPROVED** [review]
Looks good to me, should be good for a quick test-merge.

**Comment by GoldenAlpharex** [comment]
Doesn't seem to have caused any issues, let's go.

**Pull request merged**